src/USER-CAC/pair_CAC.h

```diff
@@ -0,0 +1,200 @@
+/* -*- c++ -*- ----------------------------------------------------------
+   LAMMPS - Large-scale Atomic/Molecular Massively Parallel Simulator
+   http://lammps.sandia.gov, Sandia National Laboratories
+   Steve Plimpton, sjplimp@sandia.gov
+
+   Copyright (2003) Sandia Corporation.  Under the terms of Contract
+   DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government retains
+   certain rights in this software.  This software is distributed under
+   the GNU General Public License.
+
+   See the README file in the top-level LAMMPS directory.
+------------------------------------------------------------------------- */
+
+#ifdef PAIR_CLASS
+
+PairStyle(CAC,PairCAC)
+
+#else
+
+#ifndef LMP_PAIR_CAC_H
+#define LMP_PAIR_CAC_H
+
+
+#include "pair.h"
+#include <vector>
+#include <stdint.h>
+#include "memory.h"
+#include "asa_user.h"
+using namespace std;
+
+namespace LAMMPS_NS {
+
+class PairCAC : public Pair {
+ public:
+	double cutmax;                // max cutoff for all elements
+  asacg_parm *cgParm;
```

Comment:
these need to be PIMPL-ified. that is, rather than accessing the data types directly as members of the PairCAC class, make them part of a standalone struct and then only store a pointer to that struct in the header. That allows using a forward declaration and thus bypasses the need to include "asa_user.h"